### PR TITLE
Improve a way of handling inactive app in the channel

### PR DESCRIPTION
**Pull request by korycins**
Improve a way of handling app inactive in the channel.
- add validations to webhook handler to confirm that taxjar configuration is provided & it is active
- add valdiation to not be able saving Taxjar configuration as an active, without providing API key

### CODE REVIEW

File: backend/utils.ts [hunk 1]
```diff
@@ -23,8 +19,30 @@ export const getTaxJarConfig = async (
...
-    apiKey: channelSettings.apiKey,
+    apiKey: channelSettings?.apiKey || "",
     sandbox: channelSettings?.sandbox || true,
+    active: channelSettings?.active || false,
   };
   return taxJarConfig;
 };
+
+export const taxJarConfigIsValidToUse = (taxJarConfig: TaxJarConfig) => {
+  let message = "";
```
Comment (korycins) [inline backend/utils.ts]: Any idea if I can do this better? Assuming that in the future I could have more checks here (`if`s)

File: backend/metaHandlers.ts [hunk 2]
```diff
@@ -111,3 +112,27 @@ export const fetchChannelsSettings = async (
     ? prepareResponseFromMetadata(privateMetadata, channelSlugs, false)
     : null;
 };
+
+export const validateConfigurationBeforeSave = (
+  channelsConfiguration: ConfigurationPayload
+) => {
+  const activeWithoutApiKeys = Object.entries(channelsConfiguration)
+    .filter(
+      ([_, configuration]) =>
```
Comment (bmigirl) [inline backend/metaHandlers.ts]: I think you can even remove `_`
```suggestion
      ([, configuration]) =>
```

File: backend/metaHandlers.ts [hunk 3]
```diff
@@ -111,3 +112,27 @@ export const fetchChannelsSettings = async (
...
+        configuration.active &&
+        (!configuration.apiKey || configuration.apiKey === PLACEHOLDER)
+    )
+    .map(([channelSlug, _]) => channelSlug);
```
Comment (bmigirl) [inline backend/metaHandlers.ts]:
```suggestion
    .map(([channelSlug]) => channelSlug);
```

File: backend/utils.ts (hunk 1 above)
Comment (bmigirl) [reply backend/utils.ts]: I'd say 
```
export const taxJarConfigIsValidToUse = (taxJarConfig: TaxJarConfig) => {
  const defaultResponse = {
    isValid: true,
    status: 200,
  };

  if (!taxJarConfig.active) {
    console.log("TaxJar is not active.");

    return {
      ...defaultResponse,
      message: "TaxJar is not active.",
      isValid: false,
    };
```

etc.

**Review by bmigirl — CHANGES_REQUESTED** [review]

**Pull request merged**